File: LAB 7/2.m
clc; clear;

P = [1; 2; 3; 1];

Sx = 2; Sy = 3; Sz = 4;
S = [Sx 0  0  0;
     0  Sy 0  0;
     0  0  Sz 0;
     0  0  0  1];

P_new = S * P;

disp('Scaled Point:');
disp(P_new);

File: LAB 7/2.py
import numpy as np

P = np.array([1, 2, 3, 1])

S = np.array([
    [2, 0, 0, 0],
    [0, 3, 0, 0],
    [0, 0, 4, 0],
    [0, 0, 0, 1]
])

P_new = S @ P
print("Scaled Point:", P_new)